src/uniref/adapters/uniRefConverter.ts

```diff
@@ -0,0 +1,77 @@
+enum GeneOntologyAspect {
+  FUNCTION = 'Function',
+  PROCESS = 'Process',
+  COMPONENT = 'Component',
+}
+
+enum MemberIdType {
+  UNIPROTKB = 'UniProtKb',
+  UNIPARC = 'UniParc',
+}
+
+enum EntryType {
+  UniRef100 = 'UniRef100',
+  UniRef90 = 'UniRef90',
+  UniRef50 = 'UniRef50',
+}
+
+type GeneOntologyEntry = {
+  ancestors: GeneOntologyEntry[];
+  aspect: GeneOntologyAspect;
+  name: string;
+  id: string;
+} | null;
```

Comment:
Not sure about that, because lower in this file you can have an array of those, or for its own ancestors. Which means `[null, null, null, null]` is valid. Shouldn't you just put `goTerms?: GeneOntologyEntry[]`, or `goTerms: GeneOntologyEntry[] | null`?

Reply:
So I had a chat with Leo yesterday and for UniRef they will return `ancestors: [null]`, always. But the ancestors is actually a nested array of `GeneOntologyEntry`, at least in theory. So, sticking to the data, we should defined it as `ancestors: null[]` but that just looks odd. That's why I made the whole thing nullable. And although we won't ever get `[null, null,...]` the data that we are going to get is `[null]`, so not ideal really.